packages/headless/src/features/did-you-mean/did-you-mean-slice.ts

```diff
@@ -23,12 +24,19 @@ export const didYouMeanReducer = createReducer(
         state.wasCorrectedTo = '';
       })
       .addCase(executeSearch.fulfilled, (state, action) => {
-        state.queryCorrection =
-          action.payload.response.queryCorrections[0] || emptyCorrection();
+        if (action.payload.response.queryCorrections[0]) {
+          state.queryCorrection = action.payload.response.queryCorrections[0];
```

Comment:
Related to the above comment, could we add `state.originalQuery = ...` here and remove the `didYouMeanCorrectionReceived` case?